syntax = "proto3";

service VideoStream
{
	/*
	 * Retrieved a specific video segment base on the request.
	 */
	rpc GetSegment (VideoSegmentRequest) returns (stream VideoSegmentResponse);
}

/*
 * Message to request a video segment.
 */
message VideoSegmentRequest
{
	uint32 _unused = 1;
}


/*
 * Message to respond with a video segment.
 */
message VideoSegmentResponse
{
	bytes chunk_data = 1;
	uint32 chunk_number = 2;
}
